src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -791,7 +791,20 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
             associatedSyntaxNode = associatedSyntaxNode ?? declarator;
 
             // Check for variable declaration errors.
-            bool hasErrors = this.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
+            Binder nameConflictChecker = this;
+
+            // Step out of the PatternVariableBinder for locals declared in variable declaration statement
+            if (this is PatternVariableBinder)
+            {
+                CSharpSyntaxNode parent;
+                if ((parent = declarator.Parent)?.Kind() == SyntaxKind.VariableDeclaration &&
```

Comment:
How could the test in this if statement fail? A local declaration embedded in another statement (e.g. using statement)? Is that what the included test is covering?